src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/DictionaryDefaultConverter.cs

```diff
@@ -104,7 +104,7 @@ protected static JsonConverter<TValue> GetValueConverter(ref WriteStack state)
                         // Read the value and add.
                         reader.ReadWithVerify();
                         TValue element = elementConverter.Read(ref reader, typeof(TValue), options);
-                        Add(element, options, ref state);
+                        Add(element!, options, ref state);
```

Comment:
If you remove `[return: MaybeNull]` from https://github.com/dotnet/runtime/pull/35826/files#diff-56b52018b267905bd21daf4c7615b6c7R108 you can then remove this `!`.

Reply:
Annotating the converter's `Read` method was part of the feedback from API review - https://github.com/dotnet/runtime/issues/34439#issuecomment-622040198, since now it could return `null` (since we can now pass `JsonTokenType.Null` to it).